BTrees/tests/common.py

```diff
@@ -784,6 +876,24 @@ def testPop(self):
 class BTreeTests(MappingBase):
     # Tests common to all BTrees
 
+    def _getTargetClass(self):
+        # Most of the subclasses override _makeOne and not
+        # _getTargetClass, so we can get the type that way.
+        if type(self)._makeOne is not BTreeTests._makeOne:
+            return type(self._makeOne())
+
+        # Derive the class from the test case name, if not
+        # overridden
+        name = self.__class__.__name__
+        type_name = name[:-4]
```

Comment:
Good idea. 

Actually, that whole branch isn't used; I thought I was going to refactor some things to use it, but that turned out to be (1) too much work and (2) out of scope for this PR. So I should just remove it.